core/src/main/java/org/apache/struts2/conversion/StrutsTypeConverterCreator.java

```diff
@@ -37,24 +36,22 @@ public void setObjectFactory(ObjectFactory objectFactory) {
     }
 
     public TypeConverter createTypeConverter(String className) throws Exception {
-        Object obj = objectFactory.buildBean(className, null);
-        return getTypeConverter(obj);
-    }
+        Class<?> classInstance = objectFactory.getClassInstance(className);
 
-    public TypeConverter createTypeConverter(Class<?> clazz) throws Exception {
-        Object obj = objectFactory.buildBean(clazz, null);
-        return getTypeConverter(obj);
+        if (TypeConverter.class.isAssignableFrom(classInstance)) {
```

Comment:
I think this if-else can be replaced with `return createTypeConverter(classInstance)`

Reply:
👍